cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/AccessTokenImpl.java

```diff
@@ -0,0 +1,46 @@
+package org.jasig.cas.ticket;
+
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.authentication.principal.Service;
+
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.DiscriminatorValue;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+/**
+ * An OAuth access token implementation.
+ *
+ * @author Jerome Leleu
+ * @since 4.3.0
+ */
+@Entity
+@Table(name="OAUTH_TOKENS")
+@DiscriminatorColumn(name="TYPE")
+@DiscriminatorValue(AccessToken.PREFIX)
+public class AccessTokenImpl extends OAuthCodeImpl implements AccessToken {
```

Comment:
Why is it that OAuth functionality is leaking its components into core-tickets? Should these not be part of some sort of oauth module?